src/qtism/runtime/tests/Utils.php

```diff
@@ -86,13 +89,24 @@ public static function isResponseValid(QtiDatatype $response = null, ResponseVal
 
             $patternMask = OperatorUtils::prepareXsdPatternForPcre($patternMask);
 
+            $isMaxEntryRestriction = preg_match(self::MAX_ENTRY_RESTRICTION_PATTERN, $patternMask, $matches)
+                && self::isSingleMatchGroup($patternMask);
+
             foreach ($values as $value) {
                 $result = @preg_match($patternMask, $value);
 
                 if ($result === 0) {
                     return false;
                 } elseif ($result === false) {
-                    throw new RuntimeException(OperatorUtils::lastPregErrorMessage());
+                    if ($isMaxEntryRestriction) {
+                        [$splitPattern, $min, $max] = self::extractMaxEntryRestrictionsRestrictions($matches);
+                        $entries = count(preg_split("/$splitPattern/", $response)) - 1;
```

Comment:
I'm a little confused. Shouldn't this be splitting `$value`? I mean, `$response`'s value doesn't really change in this loop.

Reply:
Defenetly, but why it works and pass tests? o_O. Will figure it out.

Reply:
That's probably because `QtiString` has a `__toString()` method ;)

Reply:
i think it shoul be something like that
```suggestion
                        $entries = count(preg_split("/$splitPattern/", $value)) - 1;
```

Reply:
It seems to me as 
$values is: `[$response->getValue()]` that's why it works

Reply:
But `$values` have nothing to do with this line ;)

Reply:
https://github.com/oat-sa/qti-sdk/blob/bfe9caaff03a65051d24d75aa3409bf56d27dffa/src/qtism/common/datatypes/QtiString.php#L95-L98
no more mistery

Reply:
i was added a test case with container but if to be honest i don't know how it can be broken cause all this pattern checks  still works with that `toString` magic of container https://github.com/oat-sa/qti-sdk/blob/c78c8185dc756d52a9f4d13e89057ec09655a7aa/src/qtism/runtime/common/Container.php#L215-L245

probably i miss something or  we can rework that loop pattern check